### PR TITLE
Budget app

### PR DESCRIPTION
**Pull request by yodit93**
### In this project, I created a mobile app with the following features
- [ ] Splash screen with
- simple page with the name of the app 
- and links to the sign-up and log in pages in which
> The user can register in the app with full name, email, and password,
> The user can log into the app using email and password.
- [ ] added authentication and authorization
- If the user is not logged in, they can't access all other pages.
- [ ] created a home page (categories page) in which
- for each category, the user can see their name, icon and the total amount of all the transactions that belong to that category.
- when the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
- there is a button "add a new category" at the bottom that brings the user to the page to create a new category.
- [ ] transactions page
- for a given category, the list of transactions is presented, ordered by the most recent.
- at the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
- there is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
- when the user clicks on the "Back" button (<), the user navigates to the home page.
- [ ] Add a new category page
the user fills out a form to create a new category, indicating their name and icon
the user clicks (or taps) the "Create category" button to create the new category, and is taken to the home page on success.
when the user clicks on the "Back" button (<), the user navigates to the home page.
- [ ] Add a new transaction" page
- the user fills out a form to create a new transaction with:
> name 
> amount
> categories
- the user clicks (or taps) the "Create Transfer" button to create the new transaction, and is taken to the transactions page for that category.
- when the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
- [ ] added unit, and integration tests 
- [ ] deployed the project on render

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb
Comment (KDlamini) [inline app/views/devise/registrations/new.html.erb:1]: Well done on your Devise authentication implementation 👏🏾💯

Please note that on this milestone you are required to follow the exact design from [the design guideline](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=). This means that the layout composition of the registration page should look exactly like this:

image

All the page designs are found [here ](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=)on the Behance website. You can take a look at them for reference.

Points to consider:

- There must be a navbar with the background color `#3778c2` that has a back arrow that navigates back to the splash screen, a "REGISTER" heading, and a "next" button on the right that serves as the "sign up" button.
- OPTIONALLY you can have the "Upload picture" field as in the design although adding the functionality is OPTIONAL. 
- The "Full name", "Email", and "password fields should not have labels but placeholders. Also, pay attention to the borders there are no right and left borders and the border color is `#F4F4F4`.

Your layout:
image

File: app/views/devise/sessions/new.html.erb
Comment (KDlamini) [inline app/views/devise/sessions/new.html.erb:1]: The same thing applies to the login page. Kindly refer to the [Behance website](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=) for reference.

The layout composition of the login page should look exactly like this:

image

Your login page:

image

File: app/views/categories/index.html.erb
Comment (KDlamini) [inline app/views/categories/index.html.erb:1]: Well done on nailing your categories page. 👏🏾🔥 

✔️ When the user logs in, they are presented with the categories page.
✔️ For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
✔️When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category
✔️There is a button "add a new category" at the bottom that brings the user to the page to create a new category

image

File: app/views/transfers/index.html.erb
Comment (KDlamini) [inline app/views/transfers/index.html.erb:1]: Well done on your transaction page implementation. 👏🏾💯 

image

✔️ For a given category, the list of transactions is presented ordered by the most recent
✔️ At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category)
✔️ There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction
✔️ When the user clicks on the "Back" button (<), the user navigates to the home page

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @yodit93 🙋‍♀️,  

Great work so far! 💯 
You have done a great job creating a working Budget app using Rails 7. 🎊 🥳 
Your project is really impressive. You have done a fantastic job!! 🥇
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞

✔️  Professional pull request title and description
✔️  Well documented README
✔️  There's user authentication, register and log in so that the data is private to them
✔️ Well-written unit and integration specs
✔️  Model validations are present
✔️  Controllers, Migrations, Views, and Routes are set up correctly

## Video Presentation

✔️ The is a recorded presentation video presenting the features of the project.
✔️ The maximum length of the video is 5 minutes.
✔️ The video contains a clear and objective description of the project in professional English. The topic and purpose of the project are made clear.
✔️ The video contains a demo of the project, highlighting all the project features.
✔️  The video contains a highlight about a piece of the code the student finds interesting or is proud of making.

## Required Changes ♻️

**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/transfers/index.html.erb
Comment (yodit93) [reply app/views/transfers/index.html.erb:1]: Thank you

File: app/views/categories/index.html.erb
Comment (yodit93) [reply app/views/categories/index.html.erb:1]: Thanks

File: app/views/devise/sessions/new.html.erb
Comment (yodit93) [reply app/views/devise/sessions/new.html.erb:1]: Ok, I will do it. Thanks 👍

File: app/views/devise/registrations/new.html.erb
Comment (yodit93) [reply app/views/devise/registrations/new.html.erb:1]: Thank you for your commitment to give these detailed suggestions 👌

**Comment by tugsem** [comment]
## STATUS: APPROVED ✔️ 
Hi @yodit93 , 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

Alt Text
Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**